custom/icds_reports/management/commands/run_custom_data_pull.py

```diff
@@ -0,0 +1,81 @@
+import os
+import zipfile
+
+from django.conf import settings
+from django.core.management.base import (
+    BaseCommand,
+    CommandError,
+)
+from django.db import connections
+
+from custom.icds_reports.const import CUSTOM_DATA_PULLS
+
+
+class Command(BaseCommand):
+    help = "Dump data from a pre-defined query for ICDS data pull requests"
+
+    def add_arguments(self, parser):
+        parser.add_argument('name')
+        parser.add_argument('db_alias', choices=settings.DATABASES)
+        parser.add_argument('--month', help="format YYYY-MM-DD")
+        parser.add_argument('--location_id')
+        parser.add_argument('-s', '--skip_confirmation', action='store_true')
+        parser.add_argument('-l', '--log_progress', action='store_true')
+
+    def handle(self, name, db_alias, *arg, **options):
+        generated_files = []
+        month = options.get('month')
+        location_id = options.get('location_id')
+        skip_confirmation = options.get('skip_confirmation')
+        log_progress = options.get('log_progress')
+        if db_alias not in settings.DATABASES:
+            raise CommandError("Unexpected db alias")
+        if name in CUSTOM_DATA_PULLS:
+            generated_files = self.run_via_class(name, db_alias, month, location_id, skip_confirmation,
+                                                 log_progress)
+        else:
+            generated_files = self.run_via_sql_file(name, db_alias, month, location_id, skip_confirmation,
+                                                    log_progress)
+        if len(generated_files) > 1:
+            zip_file_name = "%s-DataPull.zip" % name
+            with zipfile.ZipFile(zip_file_name, mode='a') as z:
+                for generated_file in generated_files:
+                    z.write(generated_file)
+                    os.remove(generated_file)
+            return zip_file_name
+        else:    
+            return generated_files[0] if generated_files else None
+
+    def run_via_class(self, slug, db_alias, month, location_id, skip_confirmation, log_progress):
+        data_pull_class = CUSTOM_DATA_PULLS[slug]
+        data_pull = data_pull_class(db_alias, month=month, location_id=location_id)
```

Comment:
isn't `data_pull_class` not guaranteed to take a `month` and `location_id`?

Reply:
Depends on the data pull, a bare/base data pull might not need any argument, or in some cases would just need a month and in some cases both month and location.
The expectations are declared in the classes inherited from `BaseDataPull` like `MonthlyDataPull` relying only on month and `LocationAndMonthBasedDataPull` needing location along with month.

Reply:
No, it would get absorbed in `kwargs`